packages/graphql-persisted/src/tests/koa-middleware.test.ts

```diff
@@ -0,0 +1,170 @@
+import faker from 'faker';
+import {Header} from '@shopify/network';
+import {createMockContext} from '@shopify/jest-koa-mocks';
+import {
+  Cache,
+  CacheMissBehavior,
+  createPersistedGraphQLMiddleware,
+} from '../koa-middleware';
+
+describe('persistedGraphQLMiddleware', () => {
+  it('only calls next when the request does not look like a persisted GraphQL query', async () => {
+    const ctx = createContext({persisted: false});
+    const cache = createCache();
+    const spy = jest.fn();
+    const persistedGraphQLMiddleware = createPersistedGraphQLMiddleware({
+      cache,
+    });
+
+    await persistedGraphQLMiddleware(ctx, spy);
+
+    expect(spy).toHaveBeenCalled();
+    expect(cache.get).not.toHaveBeenCalled();
+    expect(cache.set).not.toHaveBeenCalled();
+  });
+
+  it('does nothing if there is a persisted query body and explicit GraphQL query', async () => {
```

Comment:
Not a big fan of this test title because you are asserting on the same things as the previous test, which is not nothing.